### DISCUSSION TITLE
Get Data Year in Form Login to use in all of resource page

### DISCUSSION THREAD

**Discussion by erdowaka** [filamentphp/filament#11831] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

how can i implement this form year to set, default value in query builder ? 
this my screen shoot :
![image_2024-03-13_113005066](https://github.com/filamentphp/filament/assets/32562893/77b4ef0d-4bc0-4739-87c4-54f5f89db386)
![image_2024-03-13_113021724](https://github.com/filamentphp/filament/assets/32562893/ee666efe-d488-4e59-958f-d1ddae8f27de)
and this my resource how i implement in this resource ? 
![image_2024-03-13_113317183](https://github.com/filamentphp/filament/assets/32562893/1773c0a2-c085-49dd-823c-ff7ec05e1cb3)
i hope any one can help me for this question, thanks

**Comment by zeeshantariq08** [comment]
make a static function and call like 
`static::getYearComponents();`
```
protected static function getYearComponents()
    {
      
    }
```

**Reply by erdowaka** [reply]
okay, and how i call it ?

**Comment by leandrocfe** [comment]
```php
Select::make('year')
    ->options([
        '2022' => '2022',
        '2023' => '2023',
        '2024' => '2024',
        '2025' => '2025',
    ])
    ->required()
    ->live()
    ->default('2023')
    ->afterStateUpdated(function (string $state) {
        session()->put('user_year', $state);
    })
```

```php
return $table
    ->modifyQueryUsing(function (Builder $query) {
        $query->where('year', session()->get('user_year') ?? '2023');
    })
```

You can also use a scope
https://laravel.com/docs/11.x/eloquent#query-scopes

**Reply by erdowaka** [reply, marked as answer]
can i make this form in login page ? and how i can acces the data in resource page ?

**Reply by erdowaka** [reply]
ok done, i found that method i using session instead for handle them, any better method for handling that ?
